test/test_katello/test_deb_tracer.py

```diff
@@ -0,0 +1,73 @@
+import unittest2 as unittest
+
+# sys.path.append(os.path.join(os.path.dirname(__file__), '../../src/'))
+# from katello.tracer import upload_tracer_profile, query_affected_apps
+# from katello.constants import YUM, ZYPPER
+from katello.deb_tracer import needrestart, has_needrestart
+import subprocess
+
+from mock import patch, Mock
+
+import pytest
+
+class TestNeedrestart(unittest.TestCase):
+    tracer_output = """ perl: warning: Setting locale failed.
+perl: warning: Please check that your locale settings:
+	LANGUAGE = (unset),
+	LC_ALL = (unset),
+	LC_ADDRESS = "fr_FR.UTF-8",
+	LC_NAME = "fr_FR.UTF-8",
+	LC_MONETARY = "fr_FR.UTF-8",
+	LC_PAPER = "fr_FR.UTF-8",
+	LC_IDENTIFICATION = "fr_FR.UTF-8",
+	LC_TELEPHONE = "fr_FR.UTF-8",
+	LC_MEASUREMENT = "fr_FR.UTF-8",
+	LC_TIME = "fr_FR.UTF-8",
+	LC_NUMERIC = "fr_FR.UTF-8",
+	LANG = "en_US.UTF-8"
+    are supported and installed on your system.
+perl: warning: Falling back to a fallback locale ("en_US.UTF-8").
+NEEDRESTART-VER: 3.5
+NEEDRESTART-KCUR: 5.10.0-7-amd64
+NEEDRESTART-KEXP: 5.10.0-10-amd64
+NEEDRESTART-KSTA: 3
+NEEDRESTART-SVC: cron.service
+NEEDRESTART-SVC: dbus.service
+NEEDRESTART-SVC: getty@tty1.service
+NEEDRESTART-SVC: ifup@ens160.service
+NEEDRESTART-SVC: rsyslog.service
+NEEDRESTART-SVC: ssh.service
+NEEDRESTART-SVC: systemd-journald.service
+NEEDRESTART-SVC: systemd-logind.service
+NEEDRESTART-SVC: systemd-timesyncd.service
+NEEDRESTART-SVC: unattended-upgrades.service
+NEEDRESTART-SVC: user@0.service """
```

Comment:
There should also be some data sets, in which no service restart and no reboot need to happen.